// Copyright (c) 2022 Intel Corporation
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.ratls";
option java_outer_classname = "RATLSProto";
option objc_class_prefix = "HLW";

package ratls;

// The GRPC_RATLS secret service definition.
service GrSecret {
  // Sends a greeting
  rpc GetSecret (SecretRequest) returns (SecretReply) {}
}

// The request message containing the request's name.
message SecretRequest {
  string name = 1;
}

// The response message containing the secret (base64 encoded string)
message SecretReply {
  string secret = 1;
}
